testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/web/cache/offload/infinispan/Infinispan2ReplicasCustomCertificateService.java

```diff
@@ -0,0 +1,137 @@
+import org.infinispan.v2alpha1.Cache;
+import org.jboss.intersmash.application.openshift.OpenShiftApplication;
+import org.jboss.intersmash.application.operator.InfinispanOperatorApplication;
+import org.jboss.intersmash.tests.wildfly.web.cache.offload.infinispan.util.InfinispanCommandLineBasedKeystoreGenerator;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Application descriptor that represents an Infinispan/Red Hat Data Grid service deployed by the related Operator, and
+ * having two replicas.
+ */
+public class Infinispan2ReplicasCustomCertificateService implements InfinispanOperatorApplication, OpenShiftApplication {
+	public static final String INFINISPAN_APP_NAME = "infinispan";
+	public static final String INFINISPAN_CUSTOM_CREDENTIALS_USERNAME = "foo";
+	public static final String INFINISPAN_CUSTOM_CREDENTIALS_PASSWORD = "bar";
+	public static final String INFINISPAN_CUSTOM_CREDENTIALS_SECRET_NAME = "infinispan-custom-credentials-secret";
+	private static final Secret INFINISPAN_CUSTOM_CREDENTIALS_SECRET = buildInfinispanCustomCredentialsSecret();
+	public static final String TLS_KEYSTORE_SECRET_NAME = String.format("%s-custom-tls-secret", INFINISPAN_APP_NAME);
+	public static final String TLS_CERTIFICATE_SECRET_NAME = String.format("%s-custom-cert-secret", INFINISPAN_APP_NAME);
+	public static final String TLS_CERTIFICATE_FILE_NAME = String.format("%s.crt", INFINISPAN_APP_NAME);
+	public static final String STOREPASS = "1234PIPPOBAUDO";
+	public static final String KEYALIAS = "server";
+
+	protected Infinispan infinispan;
+	protected final List<Secret> secrets = new ArrayList<>();
+
+	private static String getEncodedIdentitiesSecretContents(InputStream is) {
+		try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
+			IOUtils.copy(is, os);
+			return Base64.getEncoder().encodeToString(os.toByteArray());
+		} catch (IOException e) {
+			throw new IllegalArgumentException("Could not process data stream", e);
+		}
+	}
+
+	private static Secret buildInfinispanCustomCredentialsSecret() {
+		try (InputStream is = Infinispan2ReplicasCustomCertificateService.class.getResourceAsStream("identities.yaml")) {
+			return new SecretBuilder()
+					.withNewMetadata().withName(INFINISPAN_CUSTOM_CREDENTIALS_SECRET_NAME).endMetadata()
+					.withData(Map.of("identities.yaml", getEncodedIdentitiesSecretContents(is)))
+					.build();
+		} catch (IOException e) {
+			throw new UnsupportedOperationException("Cannot create a custom credentials secret from identities.yaml");
+		}
+	}
+
+	public Infinispan2ReplicasCustomCertificateService() throws IOException {
+		final InfinispanCommandLineBasedKeystoreGenerator.InfinispanCertificate infinispanCertificate =
+				InfinispanCommandLineBasedKeystoreGenerator.generateInfinispanCertificate(
```

Comment:
Just use the existing utility provided by Intersmash, i.e. [CommandLineBasedKeystoreGenerator.java](https://github.com/Intersmash/intersmash/blob/main/core/src/main/java/org/jboss/intersmash/util/CommandLineBasedKeystoreGenerator.java)